File: OOps/Inheritance.py
# parent class with constructor named as Vehicle
class Vehicle:
    def __init__(self, color):
        self.color = color
    
    def honk(self):
        print("Beep! Beep!")


# child class named as Car which inherits from Vehicle

class Car(Vehicle):
    def __init__(self, color, speed):
        # call the constructor of the parent class using super
        super().__init__(color)
        self.speed = speed
    
    def accelerate(self):
        self.speed += 10
        

my_car = Car("Red", 0)
my_car.honk()

File: OOps/Inheritance.ts
// parent class with constructor named as Vehicle
class Vehicle{
    color:string;

    constructor(color:string){
        this.color = color
    }
    honk():void{
        console.log("Beep! Beep!")
    }
}

// child class named as Car
class Car extends Vehicle{
    speed:number;

    constructor(color:string, speed:number){
        // Call the constructor of the parent class (Vehicle) using super
        super(color)
        this.speed = speed
    }

    accelerate():void{
        this.speed += 10
    }
}

const MyCar = new Car("Red", 0)
MyCar.honk()